kvbc/test/sparse_merkle_storage/serialization_unit_test.cpp

```diff
@@ -108,18 +108,47 @@ TEST(key_manipulator, non_provable_key) {
 
 TEST(key_manipulator, extract_block_id_from_non_provable_key) {
   const auto non_prov_key = Sliver{"non-provable"};
-  const auto key = DBKeyManipulator::genNonProvableBlockDbKey(defaultBlockId, non_prov_key);
+  const auto key = DBKeyManipulator::genNonProvableDbKey(defaultBlockId, non_prov_key);
   const auto expected_block_id = DBKeyManipulator::extractBlockIdFromNonProvableKey(key);
   ASSERT_EQ(expected_block_id, defaultBlockId);
 }
 
 TEST(key_manipulator, extract_key_from_non_provable_key) {
   const auto non_prov_key = Sliver{"non-provable"};
-  const auto key = DBKeyManipulator::genNonProvableBlockDbKey(defaultBlockId, non_prov_key);
+  const auto key = DBKeyManipulator::genNonProvableDbKey(defaultBlockId, non_prov_key);
   const auto expected_key = DBKeyManipulator::extractKeyFromNonProvableKey(key);
   ASSERT_EQ(expected_key, non_prov_key);
 }
 
+TEST(key_manipulator, non_provable_stale_key) {
+  const auto non_prov_key = Sliver{"non-provable"};
+  const auto block_key = DBKeyManipulator::genNonProvableDbKey(defaultBlockId, non_prov_key);
```

Comment:
I think `block_key` doesn't reflect a `block` key. Maybe can be named differently.